cool-core/src/main/java/com/nus/cool/model/CoolModel.java

```diff
@@ -232,8 +228,17 @@ public static String[] listCubes(String path) {
     return localRepoPara.list();
   }
 
-  public synchronized boolean isCubeLoaded(String cube) {
-    return this.cubeStore.containsKey(cube);
+  /**
+   * check whether the cube ls loaded and is the latest.
+   *
+   * @param cube the name of the cube
+   * @return whether the cube ls loaded and is the latest
+   * @throws IOException error
+   */
+  public synchronized boolean islatestCubeLoaded(String cube) throws IOException {
+    // loaded and latest
+    return this.cubeStore.containsKey(cube) && this.storePath.get(cube).getName().equals(
+        loadLatestVersion(cube).getName());
```

Comment:
Should change the loadLatestVersion method name to getLatestVersion, since it is not loaded.

Reply:
can